nss-day-cohort-41/react-nutshell-the-macadamias
# News edit

Description
   News feed has full CRUD functionality. News articles are sorting in chronological order showing newest first. The Post New and Edit features reuse the same input fields. Edit will populate the fields with existing data.

Type of change
     New feature (non-breaking change which adds functionality)
Post, Edit, Delete all functional. Discard button included on fieldset to discard changes locally, without navigation out of the section. All API calls set within NewsAPIManager.js.

Testing Instructions
You will need React-Tiny-Link installed for this to work at all. You should have it from the last PR, but if you need to reinstall, here is the npm again.

```npm install --save react-tiny-link```

All articles should be sorting chronologically with the most recent at the top. User Articles are white. Articles from friends of the user are colored cornsilk and text is italicized.

POST new article - 
   Button at the top of the newsfeed. Clicking will open the fieldset. All fields are required. Submit is inactive until at least one character is entered. If any field is left blank, a modal will alert the user to fill out all fields before submitting. A successful submit will post a new article to the database, close the edit fields and refresh the NewsFeed. You may click the Discard button at any time prior to submit and it will close the edit fields and discard anything that had been entered.

EDIT article - Available for active user generated articles only.
  Same as above. Edit button is located on the article itself, but the fields will appear in the same location as post. The fields will be pre-populated with the values from the article selected. All other logic is still in place including modal alert for empty fields.

DELETE article - Available for active user generated articles only.
  Button on the article itself, next to the Edit button. Clicking will bring up a modal to confirm or cancel the action. A confirm will remove the article from the database and close the modal. Cancel will stop the action and close the modal.

General looks -
 The section header with the Post button should remain fixed at the top of the feed. The article section itself will scroll internally.

**OPTIONAL**
Here is some sample JSON "news" data that you can paste into your database for MAXIMUM USER EXPERIENCE!!

{
      "userId": 1,
      "url": "https://www.theonion.com/cat-too-evil-to-pet-too-soft-not-to-1844415719",
      "title": "Cat stories",
      "synopsis": "Evil Cat",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 1
    },
    {
      "userId": 2,
      "url": "https://www.cnn.com/2020/07/17/politics/ruth-bader-ginsburg-cancer-chemotherapy/index.html",
      "title": "Ruth Bader Ginsburg announces cancer recurrence",
      "synopsis": "I can't believe she has to go through this again. Hang in there notorious RBG!",
      "date": "2020-07-17T17:58:57.776Z",
      "id": 2
    },
    {
      "userId": 3,
      "url": "https://www.wsmv.com/news/dale-chihuly-exhibit-opens-saturday-at-cheekwood/article_36234ba0-c7ac-11ea-a8ff-afcfdfd07699.html",
      "title": "Dale Chihuly exhibit opens Saturday at Cheekwood",
      "synopsis": "Dale Chihuly is the mastermind behind the glass blown spectacles, setting them in gardens nationwide. This month, it's Nashville's turn.",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 3
    },
    {
      "userId": 2,
      "url": "https://www.theonion.com/report-this-article-something-to-look-at-for-few-minut-1844380629",
      "title": "I totally don't think I'm wasting my life",
      "synopsis": "So True",
      "date": "2020-07-20T04:08:30.307Z",
      "id": 5
    },
    {
      "userId": 1,
      "url": "https://www.theonion.com/fisher-price-introduces-new-bungee-jumperoo-for-thrill-1844405054",
      "title": "I need this for Christmas",
      "synopsis": "I Need a baby. Stat! No, Really!",
      "date": "2020-07-21T01:38:29.345Z",
      "id": 6
    },
    {
      "userId": 1,
      "url": "https://www.theonion.com/who-said-it-kanye-west-or-an-instruction-manual-for-th-1842383749",
      "title": "I'm not sure who said it better",
      "synopsis": "Yeah, I said it!",
      "date": "2020-07-21T03:53:03.502Z",
      "id": 7
    }


Checklist:
 My code follows the style guidelines of this project
 I have performed a self-review of my own code
 I have commented my code, particularly in hard-to-understand areas
 My changes generate no new warnings or errors
 I have added test instructions that prove my fix is effective or that my feature works

Thanks for looking,
Patrick
